### PR TITLE
Various enhancements

### PR DESCRIPTION
**Pull request by niden**
- Added boxfile for nanobox integration
- Added provision to generate code coverage report
- Added scrutinizer file
- Minor changes and added tests
- Sorted methods alphabetically (public/protected/private)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,5 +5,6 @@
 #
 # https://help.github.com/articles/ignoring-files/#create-a-global-gitignore
 
+/ci
```
Comment (sergeyklay) [inline .gitignore]: Do we really need this?

File: README.md [hunk 2]
```diff
@@ -3,6 +3,9 @@
 
 [![Software License](https://img.shields.io/badge/license-BSD--3-brightgreen.svg?style=flat-square)][:license:]
 [![Build Status](https://travis-ci.org/phalcon/cli-options-parser.svg?branch=master)](https://travis-ci.org/phalcon/cli-options-parser)
+[![Scrutinizer Code Quality](https://scrutinizer-ci.com/g/phalcon/cli-options-parser/badges/quality-score.png?b=master)](https://scrutinizer-ci.com/g/phalcon/cli-options-parser/?branch=master)
+[![Code Coverage](https://scrutinizer-ci.com/g/phalcon/cli-options-parser/badges/coverage.png?b=master)](https://scrutinizer-ci.com/g/phalcon/cli-options-parser/?branch=master)
```
Comment (sergeyklay) [inline README.md]: I've added code coverage badge from https://codecov.io/gh/phalcon/cli-options-parser

File: README.md [hunk 3]
```diff
@@ -3,6 +3,9 @@
...
+[![Build Status](https://scrutinizer-ci.com/g/phalcon/cli-options-parser/badges/build.png?b=master)](https://scrutinizer-ci.com/g/phalcon/cli-options-parser/build-status/master)
```
Comment (sergeyklay) [inline README.md]: Can we have another label here. For example Scrutinizer status or something like that

File: boxfile.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      - tokenizer
+      - xml
+      - xmlwriter
+    zend_extensions:
+      - opcache
+    dev_zend_extensions:
+      add:
+        - xdebug
+      rm:
+        - opcache
```
Comment (sergeyklay) [inline boxfile.yml:41]: Actually I can realize why do we need this in current project

File: src/Parser.php [hunk 5]
```diff
@@ -18,9 +20,6 @@
  */
 class Parser
 {
-    /** @var array */
-    private $parsedCommands;
-
```
Comment (sergeyklay) [inline src/Parser.php:23]: Why? :)

**Review by sergeyklay — CHANGES_REQUESTED** [review]

**Comment by sergeyklay** [comment]
@niden In general looks good to me (except some minor issues). I just improved testing process and build environment in https://github.com/phalcon/cli-options-parser/pull/7 so this PR branch has conflicts that must be resolved. Could you please rebase

File: src/Parser.php (hunk 5 above)
Comment (niden) [reply src/Parser.php:23]: Alphabetical sort of properties and methods. Makes it a bit easier to read.

File: boxfile.yml (hunk 4 above)
Comment (niden) [reply boxfile.yml:41]: If you want to have an isolated environment where you can code or run tests then nanobox will help with that without interfering with your other installations. Similar to docker compose etc. The reason I added it was to run tests

File: README.md (hunk 3 above)
Comment (niden) [reply README.md]: `Scrutinizer Build Status`

File: .gitignore (hunk 1 above)
Comment (niden) [reply .gitignore]: This is to store the code coverage report for the CI. I can remove that from here and build the report in the root. Either is fine.

Comment (niden) [reply .gitignore]: Actually I just saw that you got the `build` directory there so this can be removed.

**Review by sergeyklay — APPROVED** [review]

**Comment by sergeyklay** [comment]
Thank you!

**Pull request merged**